src/libraries/System.Net.NameResolution/tests/FunctionalTests/HostOverridesTest.cs

```diff
@@ -0,0 +1,151 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.RemoteExecutor;
+using Xunit;
+
+namespace System.Net.NameResolution.Tests
+{
+    public class HostAliasesTest : FileCleanupTestBase
+    {
+        private const string HostsEnvironmentVariableName = "DOTNET_SYSTEM_NET_HOSTS";
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void AllHookedEntrypoints_LookupsFindExpectedData()
+        {
+            string hostsPath = CreateTestFile();
+            File.WriteAllText(hostsPath, $"""
+                # This is a sample HOSTS file
+                #
+                102.54.94.97     rhino.acme.com  # something
+                 38.25.63.10     x.acme.com      # something else
+                ::1              example
+
+                nonparsableip
+                .....
+                1.2..3 test
+
+                ::1  invalid$$$host$@
+                1.1.1.1 firstnameonline    secondnameonline   thirdnameonline secondnameonline
+
+                1.2.3.4          example2
+                ::2          example2
+                ::1  localhost
+                ::1  anothermappedto1
+                """);
```

Comment:
Should it test long ipv6 addresses?
Eg

2001:0db8:0000:0000:0000:8a2e:0370:7334 

2001:db8::8a2e:370:7334